### PR TITLE
New reconciler

### PR DESCRIPTION
**Pull request by ledamint**

### CODE REVIEW

File: src/ManagerBot/ManagerBot.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        this.promiseQueueMap.set(id, new PromiseQueue());
+    }
+
+    deleteChat(id) {
+        this.promiseQueueMap.delete(id);
+    }
+
+    on(event, listener, chatId) {
+        return this.bot.on(event, function(ctx) {
+            if (chatId !== undefined) {
```
Comment (oTkPoBeHuE) [inline src/ManagerBot/ManagerBot.js:19]: Add an intermediate variable with a сlear name or create a function for same purposes.

File: src/TelegramBot/TelegramBot.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    sendMessage(nodeName, chatId, data) {
+        switch (nodeName) {
+            case 'text': {
+                return this.bot.sendMessage(chatId, data.text, data);
+            }
+            case 'img': {
+                return this.bot.sendPhoto(chatId, data.src, data);
+            }
+            default: {
+                throw new Error('tag ' + nodeName + ' does not exist');
```
Comment (oTkPoBeHuE) [inline src/TelegramBot/TelegramBot.js]: 'Tag ${nodeName } does not exist.'

Comment (oTkPoBeHuE) [reply src/TelegramBot/TelegramBot.js]: 'Tag ${nodeName } does not exist. ...Instructions on what to do...... '

File: src/hooks.js [hunk 3]
```diff
@@ -13,18 +13,10 @@ function useSubscribe(func, event) {
...
-                func(ctx);
-            }
-        }
-
-        bot.on(event, handler);
+        bot.on(event, func, chat.id);
 
         return () => {
-            bot.removeListener(event, handler);
+            bot.removeListener(event, func);
```
Comment (oTkPoBeHuE) [inline src/hooks.js]: I think the name "func" is too vague. When we use Typescript, there will be a type of Function - we get a tautology.

File: src/TelegramBot/TelegramBot.js (hunk 2 above)
Comment (ledamint) [reply src/TelegramBot/TelegramBot.js]: Added

**Pull request merged**